### ISSUE TITLE
Firefox rendering performance

### ISSUE THREAD

**Issue by tearmaker** [browserquest/BrowserQuest#189]
Hi,
As I add more sprites, am I the only one having ~20fps on Firefox and 60fps on Chrome, IE, Opera ? Is there a solution ?

**Comment by ghost** [comment]
No, happens to my version of browser quest, that's why I'm rewriting it all from scratch.

**Comment by tearmaker** [comment]
I think replacing the engine by pixi.js or phaser would be perfect and solve the problem. But it's quite a long work.
There is still webgl-2D ( https://github.com/gameclosure/webgl-2d ), that is way easier to plug in the existing browserquest, but it didn't work for me (I didn't try long neither).

**Comment by rickguedes** [comment]
According to some Mozilla dev:

I believe this is because the body element has:
background-image: radial-gradient(transparent 40%, rgb(0, 0, 0) 125%), url("http://cdn.mozilla.net/browserquest/img/1/wood.png"), none;

One could also reasonably ask why that background is being repainted all the time, given that it's not changing at all as you play the game.. the game is played in a canvas, so maybe we should cache the background rendering into a layer?

**Comment by rickguedes** [comment]
After some tests, it's not related to the gradient at all. It's fault of the awfully optimized desktop renderer (it just cleans and repaint the screen completely every frame without any buffering, a major waste of resources)
